packages/block-library/src/video/tracks-editor.js

```diff
@@ -0,0 +1,305 @@
+				return (
+					<div
+						key={ index }
+						className="block-library-video-tracks-editor__track-list-track"
+					>
+						<span>{ track.label } </span>
+						<Button
+							isTertiary
+							onClick={ () => onEditPress( index ) }
+							aria-label={ sprintf(
+								/* translators: %s: Label of the video text track e.g: "French subtitles" */
+								__( 'Edit %s' ),
+								track.label
+							) }
+						>
+							{ __( 'Edit' ) }
+						</Button>
+					</div>
+				);
+			} ) }
+		</MenuGroup>
+	);
+}
+
+function SingleTrackEditor( { track, onChange, onClose, onRemove } ) {
+	const { src = '', label = '', srcLang = '', kind = DEFAULT_KIND } = track;
+	const fileName = src.startsWith( 'blob:' )
+		? ''
+		: src.substring( src.lastIndexOf( '/' ) + 1 );
+	return (
+		<div className="block-library-video-tracks-editor__single-track-editor">
+			<span className="block-library-video-tracks-editor__single-track-editor-edit-track-label">
+				{ __( 'Edit track' ) }
+			</span>
+			<span>
+				{ __( 'File' ) }: <b>{ fileName }</b>
+			</span>
+			<div className="block-library-video-tracks-editor__single-track-editor-label-language">
+				<TextControl
+					/* eslint-disable jsx-a11y/no-autofocus */
+					autoFocus
+					/* eslint-enable jsx-a11y/no-autofocus */
+					onChange={ ( newLabel ) =>
+						onChange( {
+							...track,
+							label: newLabel,
+						} )
+					}
+					label={ __( 'Label' ) }
+					value={ label }
+					help={ __( 'Title of track' ) }
+				/>
+				<TextControl
+					onChange={ ( newSrcLang ) =>
+						onChange( {
+							...track,
+							srcLang: newSrcLang,
+						} )
+					}
+					label={ __( 'Source language' ) }
+					value={ srcLang }
+					help={ __( 'Language tag (en, fr, etc.)' ) }
+				/>
+			</div>
+			<SelectControl
+				className="block-library-video-tracks-editor__single-track-editor-kind-select"
+				options={ KIND_OPTIONS }
+				value={ kind }
+				label={ __( 'Kind' ) }
+				onChange={ ( newKind ) => {
+					if ( newKind === DEFAULT_KIND ) {
+						newKind = undefined;
+					}
+					onChange( {
+						...track,
+						kind: newKind,
+					} );
+				} }
+			/>
+			<div className="block-library-video-tracks-editor__single-track-editor-buttons-container">
+				<Button
+					isSecondary
+					onClick={ () => {
+						const changes = {};
+						let hasChanges = false;
+						if ( label === '' ) {
+							changes.label = __( 'English' );
+							hasChanges = true;
+						}
+						if ( srcLang === '' ) {
+							changes.srcLang = 'en';
+							hasChanges = true;
+						}
+						if ( hasChanges ) {
+							onChange( {
+								...track,
+								...changes,
+							} );
+						}
+						onClose();
+					} }
+				>
+					{ __( 'Close' ) }
+				</Button>
+				<Button isDestructive isLink onClick={ onRemove }>
+					{ __( 'Remove track' ) }
+				</Button>
+			</div>
+		</div>
+	);
+}
+
+export default function TracksEditor( { tracks = [], onChange } ) {
+	const mediaUpload = useSelect( ( select ) => {
```

Comment:
Just noting that this can be undefined, in which case we should probably hide the button entirely right?

Reply:
Or maybe allow using external  URLs (can be for later).

Reply:
Yah not rendering anything seems like a good option for now. I updated the code.